Meta/CMake/flac_spec_tests.cmake

```diff
@@ -0,0 +1,24 @@
+if(INCLUDE_FLAC_SPEC_TESTS)
+    if (CMAKE_PROJECT_NAME STREQUAL "SerenityOS")
+        set(SOURCE_DIR "${SerenityOS_SOURCE_DIR}")
+    else()
+        set(SOURCE_DIR "${SERENITY_PROJECT_ROOT}")
+    endif()
+    set(FLAC_SPEC_TEST_GZ_URL https://github.com/ietf-wg-cellar/flac-test-files/archive/refs/heads/main.tar.gz)
+
+    set(FLAC_TEST_PATH ${CMAKE_BINARY_DIR}/Tests/LibAudio/FLAC CACHE PATH "Location of FLAC tests")
+    set(FLAC_SPEC_TEST_GZ_PATH ${FLAC_TEST_PATH}/flac-spec-testsuite.tar.gz)
+    set(FLAC_SPEC_TEST_PATH ${FLAC_TEST_PATH}/SpecTests)
+
+    if(NOT EXISTS ${FLAC_SPEC_TEST_GZ_PATH})
+        message(STATUS "Downloading the IETF CELLAR FLAC testsuite from ${FLAC_SPEC_TEST_GZ_URL}...")
+        file(DOWNLOAD ${FLAC_SPEC_TEST_GZ_URL} ${FLAC_SPEC_TEST_GZ_PATH} INACTIVITY_TIMEOUT 10)
+    endif()
```

Comment:
Let's use the `download_file` utility from `Meta/CMake/utils.cmake` instead, which handles errors. Otherwise if the file fails to download halfway through, you're left with a corrupt .tar.gz in your project.